lightning/src/ln/channelmanager.rs

```diff
@@ -1863,6 +1886,23 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
+	/// Send a spontaneous payment, which is a payment that does not require the recipient to have
```

Comment:
What if the recipient did generate an invoice with a given payment hash and signals even `payment_secret` feature bit ? If the recipient is a LDK node, I think it'll fail the keysend reception as we don't allow payment_secret + keysend_preimage ?

Maybe a bit edgy but point to be cleared up with keysend specification ? Like invoice reception implies MUST NOT pay with `keysend`?

Reply:
Right, I think the behavior here is right, we should expect a "real" payment, the spec should mention this.